### PR TITLE
Add QA with wikipedia-dpr regression

### PR DESCRIPTION
**Pull request by manveertamber**
Needed to make some modifications to run_regression.py and DataModel.java to allow for the addition of "conversion": pyserini.eval.convert_trec_run_to_dpr_retrieval_run within the regression separate from searching or evaluation. 

There were some other small modifications as well, for example not requiring qrels_root and allowing topic_reader to be specified for each topic separately.

### CODE REVIEW

**Comment by manveertamber** [comment]
Note this regression does not include the CuratedTREC dataset https://github.com/castorini/anserini/blob/master/src/main/resources/topics-and-qrels/topics.dpr.curated.test.txt because of the need to supply the conversion script with the --regex option.

File: src/main/resources/docgen/templates/wiki-dpr-bm25.template [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Topics are stored in [`src/main/resources/topics-and-qrels/`](../src/main/resources/topics-and-qrels/).
+The regression experiments here evaluate on the test set of multiple QA datasets, namely Natural Questions, TriviaQA, SQuAD, and WebQuestions.
+
+After indexing has completed, you should be able to perform retrieval as follows:
+
+```bash
+${ranking_cmds}
+```
+
+The trec format will need to be converted to DPR's .json format for evaluation:
```
Comment (lintool) [inline src/main/resources/docgen/templates/wiki-dpr-bm25.template:39]:
```suggestion
The trec format will need to be converted to DPR's JSON format for evaluation:
```

File: docs/regressions-wiki-dpr-bm25.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+# Anserini Regressions: QA with Wikipedia-DPR Corpus
```
[the remark this answers is not in the document]
Comment (manveertamber) [reply docs/regressions-wiki-dpr-bm25.md:1]: I see your point, maybe we should stick to odqa-wiki-corpus-100w-splits. Either that or something like odqa-wiki-dpr-corpus-100w-splits to keep it consistent with what we have in pyserini https://github.com/castorini/pyserini/blob/master/pyserini/prebuilt_index_info.py#L876.

**Comment by manveertamber** [comment]
> > Note this regression does not include the CuratedTREC dataset https://github.com/castorini/anserini/blob/master/src/main/resources/topics-and-qrels/topics.dpr.curated.test.txt because of the need to supply the conversion script with the --regex option.
> 
> Can you explain why the `--regex` option causes an issue?

Would need to provide --regex under something like convert_params under this topic. 
Something like:
 - name: "[DPR: CuratedTREC Test](https://github.com/facebookresearch/DPR)"
    id: dpr-curated-test
    path: topics.dpr.curated.test.txt
    topic_reader: DprJsonl
    convert_params: --regex

Would then need to modify run_regression.py to allow the conversion script to take topic-specific convert_params like this. I felt it overcomplicated things so I left it for now, but can still do it...

**Comment by lintool** [comment]
> > > Note this regression does not include the CuratedTREC dataset https://github.com/castorini/anserini/blob/master/src/main/resources/topics-and-qrels/topics.dpr.curated.test.txt because of the need to supply the conversion script with the --regex option.
> > 
> > 
> > Can you explain why the `--regex` option causes an issue?
> 
> Would need to provide --regex under something like convert_params under this topic. Something like:
> 
> * name: "[DPR: CuratedTREC Test](https://github.com/facebookresearch/DPR)"
>   id: dpr-curated-test
>   path: topics.dpr.curated.test.txt
>   topic_reader: DprJsonl
>   convert_params: --regex
> 
> Would then need to modify run_regression.py to allow the conversion script to take topic-specific convert_params like this. I felt it overcomplicated things so I left it for now, but can still do it...

Agreed. Let's circle back on this later.

File: docs/regressions-wiki-dpr-bm25.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (manveertamber) [reply docs/regressions-wiki-dpr-bm25.md:1]: Ok let's go with wikipedia-dpr-100w. So the corpus path should be /store/collections/wikipedia/wikipedia-dpr-100w

**Review by lintool — APPROVED** [review]

**Pull request merged**